tinyDA/chain.py

```diff
@@ -347,18 +384,18 @@ def sample(self, iterations, progressbar=True):
                     alpha_2 = self._get_state_independent_acceptance(proposal_link_fine)
 
                 # Perform Metropolis adjustment, and update the coarse chain
-                # to restart from the previous accepted fine link.
+                # to restart from the previous accepted fine link.                 
                 if np.random.random() < alpha_2:
                     self.chain_fine.append(proposal_link_fine)
                     self.accepted_fine.append(True)
-                    self.chain_coarse.append(self.chain_coarse[-1])
+                    self.chain_coarse.append(self.promoted_coarse[-1])
                     self.accepted_coarse.append(True)
                     self.is_coarse.append(False)
                 else:
                     self.chain_fine.append(self.chain_fine[-1])
                     self.accepted_fine.append(False)
                     self.chain_coarse.append(
-                        self.chain_coarse[-(self.subchain_length + 1)]
+                        self.chain_coarse[-(self.subchain_length+1)]
```

Comment:
It looks like you have removed some spaces here that makes it look like there is a change but there isn't,